src/mainview.cpp

```diff
@@ -12,6 +13,11 @@ using namespace ui;
 CMainView::CMainView( QWidget *pParent ) :
 	QDialog( pParent )
 {
+	CFileSystemSearchProvider provider;
+	char *installDir = new char[1048];
+	provider.GetAppInstallDir( 440000, installDir, 1048 );
```

Comment:
Why not get `SteamAppId` instead?

Reply:
Because this isn't Steam_API?

Reply:
Well that's just a system variable regardless of steam. Check main.cpp

Reply:
... the variable I deleted from main?

Reply:
I mean, steam should've stayed since I wanted to do additional stuff with it, such as a workshop uploader

Reply:
Why would the SDK Launcher need responsibility for that? seems a little excessive. I mentioned this in this post, but I doubt it'd be needed for the SDK Launcher to be responsible for uploading to the workshop, that should be it's own application. it's how all SDK Launchers I know work.

Reply:
The reason  being is that I don't like how restrictive the Steam API is. it does a lot but only under it's own conditions. requiring internet access and steam to be actively running to even use the thing. I made SAPP not just to reimplement lost functionality of the Steam API, but to keep it offline so it can be used without the need for internet access or being logged into steam, though, still requires steam to exist on your system. as it's using it's pathing.

Reply:
``

> Why would the SDK Launcher need responsibility for that? seems a little excessive. I mentioned this in this post, but I doubt it'd be needed for the SDK Launcher to be responsible for uploading to the workshop, that should be it's own application. it's how all SDK Launchers I know work.

It's still something I'd like to add regardless of what other sdk launchers do.. it's my project after all, so I can decide ideally what I do and don't want in this.

Reply:
Then for now we'll keep it at this until the Steam API is actually needed. currently, it's sitting there adding complications, I'd like to work on the other features without "Steam Is not running on this machine" every time I close my steam app and try to run the application. When the time comes to implement the workshop uploader (still ways away, provided the P2CE workshop isn't even properly implemented yet.), I'll reimplement the Steam API and use the appid variable.

Reply:
Steam is going to be merged, but it's disabled by default. The env vars will still exist regardless, and it's not much a problem with them being assigned and not used

Reply:
> Steam is going to be merged, but it's disabled by default. The env vars will still exist regardless, and it's not much a problem with them being assigned and not used

What does that mean?

Reply:
See #10, it's hidden behind a cmake option, so you can determine whether or not to have steam in the build. But the env vars are not hidden behind this option so you can ultimately use them here. I suggest you do to use it because then you only have to change 1 thing to determine the app id, rather than 2 separate lines of code.

Reply:
> See #10, it's hidden behind a cmake option, so you can determine whether or not to have steam in the build. But the env vars are not hidden behind this option so you can ultimately use them here. I suggest you do to use it because then you only have to change 1 thing to determine the app id, rather than 2 separate lines of code.

Sounds good. Merge #10 with main and I'll add the variable to the path provider.

Reply:
I have a PR open, just waiting for somebody to look over the code, I've asked them so I'm just waiting

Reply:
Asked who?

Reply:
Somebody from the P2CE team who I know is decent when it comes to code reviews.

Reply:
Does that remove the need for SAPP? or are it's properties still useful to the SDK Launcher?

Reply:
SAPP will still be useful, steam doesn't offer much in the way to find p2ce's directly I don't think. Either way, it's still useful for when steam isn't enabled in the builds

Reply:
> I have a PR open, just waiting for somebody to look over the code, I've asked them so I'm just waiting

I know you're waiting for someone at P2CE to review it, I have looked at it myself and don't see any issues with it. it obeys styling and doesn't include bad practices.

Reply:
Well that PR got merged so you can probably rebase this branch and add what I asked for since that exists now

Reply:
I merged the branch. then cleaned up after myself.

Reply:
I also added a scroll area (left over from me still working on it), it'll prevent the application from stretching into infinity and beyond the more items are added by the config.

Reply:
How come you added the icons since they are not being used?

Reply:
I'm working on that, I actually removed some code before committing, see [6242ce5606625d59e5c2c54d3b7170e0e52f362a](https://github.com/AWildErin/sdklauncher/pull/10/commits/6242ce5606625d59e5c2c54d3b7170e0e52f362a)
I am going to use them.